code/modules/mob/living/carbon/alien/alien.dm

```diff
@@ -140,3 +143,88 @@ Des: Removes all infected images from the alien.
 
 /mob/living/carbon/alien/can_hold_items()
 	return has_fine_manipulation
+
+/obj/effect/proc_holder/alien/evolve
+	name = "Evolve"
+	desc = "Evolve into a higher alien caste."
+	plasma_cost = 0
+
+	action_icon_state = "alien_evolve_larva"
+
+///Evolves the Xenomorph if it has full progress as a larva or full plasma as a humanoid xeno, evolves into one of the evolution_paths
+/obj/effect/proc_holder/alien/evolve/fire(mob/living/carbon/alien/user)
+	. = FALSE
+	if(user.handcuffed || user.legcuffed)
+
+		to_chat(user, "<span class='warning'>You cannot evolve when you are cuffed!</span>")
+		return
+	if(islarva(user))
+		var/mob/living/carbon/alien/larva/L = user
+		if(L.amount_grown < L.max_grown)	//TODO ~Carn //TODO WHAT YOU FUCK ~Fikou
+
+			to_chat(user, "<span class='warning'>You are not fully grown!</span>")
+			return
+
+	else
+		var/obj/item/organ/alien/plasmavessel/vessel = user.getorgan(/obj/item/organ/alien/plasmavessel)
+		if(vessel.storedPlasma < vessel.max_plasma)
+			to_chat(user, "<span class='warning'>You do not have enough plasma to grow!</span>")
+			return
+
+	to_chat(user, "<span class='name'>You are growing! It is time to choose a caste.</span>")
+	var/evolutions = user.evolution_paths
+	var/alien_caste = input(user, "Please choose which alien caste you shall belong to.", "Text") as null|anything in evolutions
+	if(user.incapacitated()) //something happened to us while we were choosing.
+		return
+	var/mob/living/carbon/alien/new_xeno
+	if(!alien_caste || !(alien_caste in evolutions))
+		return
+
+	if(!isturf(user.loc))
+		to_chat(user, "<span class='warning'>You can't evolve here!</span>")
+		return
+
+	switch(alien_caste)
+		if("Larva")
+			new_xeno = new /mob/living/carbon/alien/larva(user.loc)
+		if("Sentinel")
+			new_xeno = new /mob/living/carbon/alien/humanoid/sentinel(user.loc)
+		if("Drone")
+			new_xeno = new /mob/living/carbon/alien/humanoid/drone(user.loc)

```

Comment:
Could you refactor this to use typepaths instead of strings?
Stuff like `list(/mob/living/carbon/alien/humanoid/royal/praetorian)` instead of `list("Praetorian")`

Reply:
wont it look like type paths when you select the things